pkg/controller/controller.go

```diff
@@ -838,6 +857,16 @@ func (p *csiProvisioner) Provision(ctx context.Context, options controller.Provi
 		},
 	}
 
+	// Set annDeletionSecretRefName and namespace in PV object.
+	if result.provDeletionSecrets != nil {
+		klog.V(5).Infof("createVolumeOperation: set annotation [%s/%s] on pv [%s].", annDeletionProvisionerSecretRefNamespace, annDeletionProvisionerSecretRefName, pv.Name)
+		metav1.SetMetaDataAnnotation(&pv.ObjectMeta, annDeletionProvisionerSecretRefName, result.provDeletionSecrets.name)
+		metav1.SetMetaDataAnnotation(&pv.ObjectMeta, annDeletionProvisionerSecretRefNamespace, result.provDeletionSecrets.namespace)
+	} else {
+		metav1.SetMetaDataAnnotation(&pv.ObjectMeta, annDeletionProvisionerSecretRefName, "")
+		metav1.SetMetaDataAnnotation(&pv.ObjectMeta, annDeletionProvisionerSecretRefNamespace, "")
```

Comment:
Is this needed? Won't the annotation already be unset if this is a new PV?

Reply:
Setting "" to the annotations is needed for deletion path. If we don't do it here, or on provision, and these annotations are not found on deletion, the controller can't distinguish:
1. The volume doesn't require Secret for deletion
2. The volume requires Secret for deletion but provisioned with old provisioner that doesn't add the annotations

As a result, deletion of the case 1 fails if SC is deleted. By setting "", we can avoid it like [this](https://github.com/kubernetes-csi/external-provisioner/pull/713/files/d6f20f85c31b3ea94c5a0612a344a8928ce7162f#diff-243ebed2765f75e6a54f57167212fefb08c3b2a85967ad2acbc0eb78919019c1R1219-R1221).

Reply:
Thanks @mkimuram . 
@ggriffiths hope above clarifies it..

Reply:
Ah yes, meant to reply. Thanks for the clarification!